forui/lib/src/widgets/line_calendar.dart

```diff
@@ -0,0 +1,352 @@
+
+  @override
+  State<FLineCalendar> createState() => _FLineCalendarState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('selected', selected))
+      ..add(DiagnosticsProperty('epoch', epoch))
+      ..add(DiagnosticsProperty('today', today));
+  }
+}
+
+class _FLineCalendarState extends State<FLineCalendar> {
+  late ScrollController _controller = ScrollController();
+
+  @override
+  void initState() {
+    widget.selected.addListener(() => setState(() {}));
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = widget.style ?? context.theme.lineCalendarStyle;
+
+    final textScalor = MediaQuery.textScalerOf(context);
+    final dateTextSize = textScalor.scale(style.unselectedDateTextStyle.fontSize!);
+    final dayTextSize = textScalor.scale(style.unselectedDayTextStyle.fontSize!);
+    final size = dateTextSize + dayTextSize + _textSpacing + (style.heightPadding * 2);
+
+    final offset = (widget.selected.value.difference(widget.epoch).inDays - 2) * size + style.itemPadding;
+    _controller = ScrollController(initialScrollOffset: offset);
+
+    return SizedBox(
+      height: size,
+      child: ListView.builder(
+        controller: _controller,
+        scrollDirection: Axis.horizontal,
+        padding: EdgeInsets.zero,
+        // TODO: calculate width of items based on the text font size.
+        itemExtent: size,
+        itemBuilder: (context, index) => Container(
+          padding: EdgeInsets.symmetric(horizontal: style.itemPadding),
+          child: _Tile(
+            style: widget.style,
+            selected: widget.selected,
+            date: widget.epoch.add(Duration(days: index)),
+            today: widget.today,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+}
+
+class _Tile extends StatelessWidget {
+  final FLineCalendarStyle? style;
+  final ValueNotifier<LocalDate> selected;
+  final LocalDate date;
+  final bool today;
+
+  const _Tile({required this.selected, required this.date, required LocalDate today, this.style})
+      : today = date == today;
+
+  Color _style(BuildContext context, bool selected) {
+    final style = this.style ?? context.theme.lineCalendarStyle;
+    return switch ((selected, today)) {
+      (true, true) => style.selectedCurrentDateIndicatorColor,
+      (true, false) => const Color(0x00000000),
+      (false, true) => style.unselectedCurrentDateIndicatorColor,
+      (false, false) => const Color(0x00000000),
+    };
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? context.theme.lineCalendarStyle;
+    final selected = this.selected.value == date;
+
+    return FTappable.animated(
```

Comment:
The `FTappable` lacks a semantic label. Ideally it should be the date it represents. Please also mark the label with a 'TODO: Localization'.

Reply:
In addition, we need to set `FTappable.selected` to true if it is selected.